src/sage/matrix/matrix2.pyx

```diff
@@ -3868,6 +3868,38 @@ cdef class Matrix(Matrix1):
                 % (self.nrows(), self.ncols()), level=1, t=tm)
         return 'computed-smith-form', self.new_matrix(nrows=len(basis), ncols=self._ncols, entries=basis)
 
+    def _right_kernel_matrix_over_integer_mod_ring(self):
+        r"""
+        Returns a pair that includes a matrix of basis vectors
+        for the right kernel of ``self``.
+
+        OUTPUT:
+
+        Returns a pair. First item is the string 'computed-pari-matkermod'
+        that identifies the nature of the basis vectors.
+
+        Second item is a matrix whose rows are a basis for the right kernel,
+        over the integer modular ring, as computed by :pari:`matkermod`.
+
+        EXAMPLES::
+
+            sage: A = matrix(Zmod(24480), [[1,2,3,4,5],[7,7,7,7,7]])
+            sage: result = A._right_kernel_matrix_over_integer_mod_ring()
+            sage: result[0]
+            'computed-pari-matkermod'
+            sage: P = result[1]; P
+            [    1 24478     1     0     0]
+            [    2 24477     0     1     0]
+            [    3 24476     0     0     1]
+            sage: A*P.transpose() == 0
+            True
+        """
+        R = self.base_ring()
+        pariM = self.change_ring(ZZ).__pari__()
```

Comment:
This seems costly as it creates a whole new matrix, which then creates another whole copy for PARI. It would be good if you could expand this to avoid this transient object. If it is too much trouble, then we can skip it for now.

Reply:
I've tried both `self.__pari__().liftint()` and copying the coefficients one by one to a new PARI matrix, and both are significantly slower than the code that is currently in the branch.

Reply:
That's a bit surprising.  Perhaps it is something that is best implemented upstream in cypari? Well, that can be done at a later time.